### PR TITLE
arm: mps: docs: note on qemu usage of boards

### LINKED ISSUE #18154: Qemu: mps2+: missing documentation
There needs to be some documentation under
-  boards/boards/arm/mps2_an385/doc
-  boards/boards/arm/mps2_an521/doc

That documentation should indicate that these are Qemu targets for ARM; how they can be used; known issues, etc.

Probably just copy-paste with trivial adjustments from some other board's QEMU documentation

See, for examples, qemu_cortex_m3

### PR DESCRIPTION
**Pull request by microbuilder**
This commit adds a note alluding to the fact that these two boards are
primarily included for use with QEMU, but have been tested on actual
hardware as well where appropriate.

Resolves #18154 .

Signed-off-by: Kevin Townsend <kevin.townsend@linaro.org>

### CODE REVIEW

**Review by ioannisg — COMMENTED** [review]
Could we say, somewhere in the overview (besides the provided note) that the board definitions are used (also) in Qemu ?

E.g. something like what we say for qemu_cortex_m3:

>This board configuration will use QEMU to emulate the TI LM3S6965 platform.

**Comment by microbuilder** [comment]
> This board configuration will use QEMU

```
In addition to enabling actual hardware usage, this board configuration can
also use QEMU to emulate the AN521 platform running on the MPS2+.
```

???

File: boards/arm/mps2_an385/doc/index.rst [hunk 1]
```diff
@@ -20,10 +20,19 @@ the following devices:
...
+also use QEMU to emulate the AN385 platform running on the MPS2+.
+
 More information about the board can be found at the `V2M MPS2 Website`_.
 
 The Application Note AN385 can be found at `Application Note AN385`_.
 
+.. note::
+   This board configuration makes no claims about its suitability for use
+   with actual MPS2 hardware systems using AN385, or any other hardware
+   system. It has been tested on actual hardware, but it's primary purpose is
```
Comment (MaureenHelm) [inline boards/arm/mps2_an385/doc/index.rst]: s/it's/its/

File: boards/arm/mps2_an521/doc/index.rst [hunk 2]
```diff
@@ -21,8 +21,17 @@ CPU and the following devices:
...
 
+In addition to enabling actual hardware usage, this board configuration can
+also use QEMU to emulate the AN521 platform running on the MPS2+.
+
 More information about the board can be found at the `MPS2 FPGA Website`_.
 
+.. note::
+   This board configuration makes no claims about its suitability for use
+   with actual MPS2 hardware systems using AN521, or any other hardware
+   system. It has been tested on actual hardware, but it's primary purpose is
```
Comment (MaureenHelm) [inline boards/arm/mps2_an521/doc/index.rst]: s/it's/its/

**Review by MaureenHelm — DISMISSED** [review]
Minor grammar edit, but otherwise looks fine

**Review by ioannisg — APPROVED** [review]

File: boards/arm/mps2_an521/doc/index.rst (hunk 2 above)
Comment (ioannisg) [reply boards/arm/mps2_an521/doc/index.rst]: agree

**Review by ioannisg — COMMENTED** [review]
Thanks for the update @microbuilder , pls, address @MaureenHelm 's fb

File: boards/arm/mps2_an385/doc/index.rst [hunk 3]
```diff
@@ -20,10 +20,19 @@ the following devices:
...
+   for use with qemu and unit tests.
```
Comment (dbkinder) [inline boards/arm/mps2_an385/doc/index.rst]: s/qemu/QEMU/

File: boards/arm/mps2_an521/doc/index.rst [hunk 4]
```diff
@@ -21,8 +21,17 @@ CPU and the following devices:
...
+   for use with qemu and unit tests.
```
Comment (dbkinder) [inline boards/arm/mps2_an521/doc/index.rst]: s/qemu/QEMU/

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: boards/arm/mps2_an385/doc/index.rst (hunk 1 above)
Comment (dbkinder) [reply boards/arm/mps2_an385/doc/index.rst]: FYI, ``it's`` is a contraction of ``it is``

**Review by dbkinder — DISMISSED** [review]

**Review by carlescufi — APPROVED** [review]

**Review by ioannisg — APPROVED** [review]

**Comment by ioannisg** [comment]
Funny that the commit messages are identical

**Pull request merged**